pkg/applylib/applyset/applyset.go

```diff
@@ -0,0 +1,232 @@
+//	We don't _have_ to try to apply all objects if it is taking too long.
+//
+// TODO: We re-apply every object every iteration; we should be able to do better.
+func (a *ApplySet) ApplyOnce(ctx context.Context) (*ApplyResults, error) {
+	// snapshot the state
+	a.mutex.Lock()
+	trackers := a.trackers
+	a.mutex.Unlock()
+
+	results := &ApplyResults{total: len(trackers.items)}
+
+	for i := range trackers.items {
+		tracker := &trackers.items[i]
+		obj := tracker.desired
+
+		name := obj.GetName()
+		ns := obj.GetNamespace()
+		gvk := obj.GroupVersionKind()
+		nn := types.NamespacedName{Namespace: ns, Name: name}
+
+		restMapping, err := a.restMapper.RESTMapping(gvk.GroupKind(), gvk.Version)
+		if err != nil {
+			results.applyError(gvk, nn, fmt.Errorf("error getting rest mapping for %v: %w", gvk, err))
+			continue
+		}
+		gvr := restMapping.Resource
+
+		var dynamicResource dynamic.ResourceInterface
+
+		switch restMapping.Scope.Name() {
+		case meta.RESTScopeNameNamespace:
+			if ns == "" {
+				// TODO: Differentiate between server-fixable vs client-fixable errors?
+				results.applyError(gvk, nn, fmt.Errorf("namespace was not provided for namespace-scoped object %v", gvk))
+				continue
+			}
+			dynamicResource = a.client.Resource(gvr).Namespace(ns)
+
+		case meta.RESTScopeNameRoot:
+			if ns != "" {
+				// TODO: Differentiate between server-fixable vs client-fixable errors?
+				results.applyError(gvk, nn, fmt.Errorf("namespace %q was provided for cluster-scoped object %v", obj.GetNamespace(), gvk))
+				continue
+			}
+			dynamicResource = a.client.Resource(gvr)
+
+		default:
+			// Internal error ... this is panic-level
+			return nil, fmt.Errorf("unknown scope for gvk %s: %q", gvk, restMapping.Scope.Name())
+		}
+
+		j, err := json.Marshal(obj)
+		if err != nil {
+			// TODO: Differentiate between server-fixable vs client-fixable errors?
+			results.applyError(gvk, nn, fmt.Errorf("failed to marshal object to JSON: %w", err))
+			continue
+		}
+
+		var lastApplied *unstructured.Unstructured
+		switch a.ApplyMethod {
+		case ApplyMethodServerSideApply, "":
+			applied, err := dynamicResource.Patch(ctx, name, types.ApplyPatchType, j, a.patchOptions)
+			if err != nil {
+				results.applyError(gvk, nn, fmt.Errorf("error from apply: %w", err))
+				continue
+			}
+			lastApplied = applied
+
+		case ApplyMethodCreateOrUpdate:
+			updated, err := a.createOrUpdateObject(ctx, dynamicResource, name, j)
+			if err != nil {
+				results.applyError(gvk, nn, err)
+				continue
+			}
+			lastApplied = updated
+
+		default:
+			return nil, fmt.Errorf("unknown apply method %q", a.ApplyMethod)
+		}
+
+		tracker.lastApplied = lastApplied
+		results.applySuccess(gvk, nn)
+		tracker.isHealthy = isHealthy(lastApplied)
+		results.reportHealth(gvk, nn, tracker.isHealthy)
+	}
+	return results, nil
+}
+
+func (a *ApplySet) createOrUpdateObject(ctx context.Context, resource dynamic.ResourceInterface, name string, json []byte) (*unstructured.Unstructured, error) {
```

Comment:
So I included it so that the transition from our apply in kOps would be easier.  But now you've moved that to SSA, I agree we should just remove this functionality.  I'll remove it!

Reply:
Removed!